src/store/modules/ErrorModule.ts

```diff
@@ -16,11 +18,17 @@ export default {
 
     state: (): State => ({
         error: null,
+        extraActionButton: null,
     }),
 
     mutations: {
         discardError: function(state: State): void {
             state.error = null;
+            state.extraActionButton = null;
+        },
+
+        setExtraActionButton: function(state: State, extraActionButton: Component): void {
```

Comment:
I'm not a fan of having the action button a separate thing in the state as it seems unnecessarily complex. For example, it seems to me the button is never reset, so it gets shown again for the next error too? Would it be simpler to pass the button text and the action function within a new optional fields in the R2Error object? It gets passed to error modal anyway and it would ensure the error and the action would always be in sync.

Reply:
After the changes already done, are any changes actually needed in this file or should they all be reverted?